src/UserSpaceInstrumentation/ExecuteMachineCodeTest.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <gtest/gtest.h>
+#include <sys/wait.h>
+
+#include <csignal>
+
+#include "AllocateInTracee.h"
+#include "ExecuteMachineCode.h"
+#include "MachineCode.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "UserSpaceInstrumentation/Attach.h"
+
+namespace orbit_user_space_instrumentation {
+
+TEST(ExecuteMachineCodeTest, ExecuteMachineCode) {
+  pid_t pid = fork();
+  ASSERT_TRUE(pid != -1);
+  if (pid == 0) {
+    while (true) {
+    }
+  }
+
+  // Stop the child process using our tooling.
+  ASSERT_TRUE(AttachAndStopProcess(pid).has_value());
+
+  // Allocate a small chunk of memory.
+  constexpr uint64_t kScratchPadSize = 1024;
+  ErrorMessageOr<uint64_t> address_or_error = AllocateInTracee(pid, 0, kScratchPadSize);
+  ASSERT_TRUE(address_or_error.has_value());
+  const uint64_t address = address_or_error.value();
+
+  // This code moves a constant into rax and enters a breakpoint. The value in rax is interpreted as
+  // a return value.
+  // movabs rax, 0x4242424242424242     48 b8 0x4242424242424242
+  // int 3                              cc
+  MachineCode code;
+  code.AppendBytes({0x48, 0xb8}).AppendImmediate64(0x4242424242424242).AppendBytes({0xcc});
+  ErrorMessageOr<uint64_t> result_or_error = ExecuteMachineCode(pid, address, code);
+  ASSERT_FALSE(result_or_error.has_error());
```

Comment:
<< result_or_error.error().message();

Reply:
I added this here and in the places where the assert really tests something in the scope of the test. 

In many other places I replaced the asserts with CHECK to document that the code is setting up or tearing down the test but is not really part of the thing I want to test. 

Specifically I want to avoid blowing up one liners like
`ASSERT_TRUE(AttachAndStopProcess(pid).has_value());`
or
`ASSERT_FALSE(FreeInTracee(pid, address, kScratchPadSize).has_error());` by storing a the return value.

tracking a clenaup here: http://b/185312970 There are a bunch of asserts that should be checks in the other tests.